deepchem/molnet/load_function/zinc15_datasets.py

```diff
@@ -0,0 +1,233 @@
+  are likely to bind to drug targets. ZINC15 data is currently available
+  in 2D (SMILES string) format.
+
+  MolNet provides subsets of 250K, 1M, and 10M "lead-like" compounds
+  from ZINC15. Compounds in ZINC15 are labeled by their molecular weight 
+  and LogP (solubility) values. Each compound also has information about how
+  readily available (purchasable) it is and its reactivity. Lead-like
+  compounds have molecular weight between 300 and 350 Daltons and LogP
+  between -1 and 3.5.
+
+  If `reload = True` and `data_dir` (`save_dir`) is specified, the loader
+  will attempt to load the raw dataset (featurized dataset) from disk.
+  Otherwise, the dataset will be downloaded from the DeepChem AWS bucket.
+
+  For more information on ZINC15, please see [1]_ and
+  https://zinc15.docking.org/.
+
+  Parameters
+  ----------
+  size : str (default '250K')
+    Size of dataset to download. Currently only '250K' is supported.
+  format : str (default '2D')
+    Format of data to download. 2D SMILES strings or 3D SDF files.
+  featurizer : allowed featurizers for this dataset
+    A featurizer that inherits from deepchem.feat.Featurizer.
+  transformers : List of allowed transformers for this dataset
+    A transformer that inherits from deepchem.trans.Transformer.
+  splitter : allowed splitters for this dataset
+    A splitter that inherits from deepchem.splits.splitters.Splitter.
+  reload : bool (default True)
+    Try to reload dataset from disk if already downloaded. Save to disk
+    after featurizing.
+  data_dir : str, optional (default None)
+    Path to datasets.
+  save_dir : str, optional (default None)
+    Path to featurized datasets.
+  featurizer_kwargs : dict
+    Specify parameters to featurizer, e.g. {"size": 1024}
+  splitter_kwargs : dict
+    Specify parameters to splitter, e.g. {"seed": 42}
+  transformer_kwargs : dict
+    Maps transformer names to constructor arguments, e.g.
+    {"BalancingTransformer": {"transform_x":True, "transform_y":False}}
+  zinc15_kwargs : dict
+    Specify parameters for the ZINC15 dataset. Accepted keywords are
+    'dataset_size' and 'dataset_dimension'.
+  **kwargs : additional optional arguments.
+
+  Returns
+  -------
+  tasks, datasets, transformers : tuple
+    tasks : list
+      Column names corresponding to machine learning target variables.
+    datasets : tuple
+      train, validation, test splits of data as
+      ``deepchem.data.datasets.Dataset`` instances.
+    transformers : list
+      ``deepchem.trans.transformers.Transformer`` instances applied
+      to dataset.
+
+  Notes
+  -----
+  The total ZINC dataset with SMILES strings contains hundreds of millions
+  of compounds and is over 100GB! ZINC250K is recommended for experimentation.
+
+  References
+  ----------
+  .. [1] Sterling and Irwin. J. Chem. Inf. Model, 2015 http://pubs.acs.org/doi/abs/10.1021/acs.jcim.5b00559.
+
+  Examples
+  --------
+  >> import deepchem as dc
```

Comment:
One thing I've noticed is that the `>>` style examples don't render correctly in readthedocs. Could we come up with a way to turn this into a functioning code sample so we can use the `>>>` example? Perhaps we could allow for something like a "dry-run" flag that returns a dataset of all zeros but which has features/labels of the right shape

Reply:
Ah, I see what you mean. Strange that it doesn't create newlines with the `>>` style. Maybe I can point the example to the small data set in the `tests` folder so that it will run correctly. If that doesn't work I'll try a `dry-run` flag.